scripts/validate-artifacts.ps1

```diff
@@ -164,42 +135,45 @@ if ($nic) {
-}
 
-if ($virtualMachine.properties.storageProfile.imageReference.publisher -eq "canonical") { 
-    Write-Output "`u{2705} Checked Virtual Machine OS image publisher - OK" 
-} else { 
+$nsg = ( $TemplateObject.resources | Where-Object -Property type -EQ "Microsoft.Network/networkSecurityGroups")
+if ($nsg) {
+    if ($nsg.name.Count -eq 1) { 
+        Write-Output "`u{2705} Checked if the Network Security Group resource exists - OK"
+    }  else { 
+        Write-Output `u{1F914}
+        throw "More than one Network Security Group resource was found in the VM resource group. Please delete all un-used Network Security Group resources and try again."
+    }
+} else {
     Write-Output `u{1F914}
-    throw "Virtual Machine uses OS image from unknown published. Please make sure that your script creates a VM from image with friendly name 'Ubuntu2204' and try again."
+    throw "Unable to find Network Security Group resouce. Please re-deploy the VM and try again."
 }
-if ($virtualMachine.properties.storageProfile.imageReference.offer.Contains('ubuntu-server') -and $virtualMachine.properties.storageProfile.imageReference.sku.Contains('22_04')) { 
-    Write-Output "`u{2705} Checked Virtual Machine OS image offer - OK"
+
+$sshNsgRule = ( $nsg.properties.securityRules | Where-Object { ($_.properties.destinationPortRange -eq '22') -and ($_.properties.access -eq 'Allow')} ) 
+if ($sshNsgRule)  {
+    Write-Output "`u{2705} Checked if NSG has SSH network security rule configured - OK"
 } else { 
     Write-Output `u{1F914}
-    throw "Virtual Machine uses wrong OS image. Please make sure that your script creates a VM from image with friendly name 'Ubuntu2204' and try again." 
+    throw "Unable to fing network security group rule which allows SSH connection. Please check if you configured VM Network Security Group to allow connections on 22 TCP port and try again."
 }
 
-if ($virtualMachine.properties.hardwareProfile.vmSize -eq "Standard_B1s") { 
-    Write-Output "`u{2705} Checked Virtual Machine size - OK"
+$httpNsgRule = ( $nsg.properties.securityRules | Where-Object { ($_.properties.destinationPortRange -eq '8080') -and ($_.properties.access -eq 'Allow')} ) 
+if ($httpNsgRule)  {
+    Write-Output "`u{2705} Checked if NSG has HTTP network security rule configured - OK"
 } else { 
     Write-Output `u{1F914}
-    throw "Virtual Machine size is not set to B1s. Please make sure that your script creates a VM with size B1s and try again."
+    throw "Unable to fing network security group rule which allows HTTP connection. Please check if you configured VM Network Security Group to allow connections on 8080 TCP port and try again."
 }
 
 $response = (Invoke-WebRequest -Uri "http://$($pip.properties.dnsSettings.fqdn):8080/api/" -ErrorAction SilentlyContinue) 
```

Comment:
The `Invoke-WebRequest` command checks if the web application is running. Ensure that the DNS settings and port are correctly configured to avoid connection errors.